src/pages/CommitDetailPage/CommitCoverage/routes/IndirectChangesTab/IndirectChangesTable/CommitFileDiff/CommitFileDiff.jsx

```diff
@@ -83,6 +82,30 @@ function CommitFileDiff({ path }) {
       {isCriticalFile && <CriticalFileLabel variant="borderTop" />}
       {segments?.map((segment, segmentIndex) => {
         const content = segment.lines.map((line) => line.content).join('\n')
+
+        let newDiffContent = ''
+        const lineData = []
+        if (virtualDiffRenderer) {
+          segment.lines.forEach((line, lineIndex) => {
+            newDiffContent += line.content
+
+            if (lineIndex < segment.lines.length - 1) {
+              newDiffContent += '\n'
+            }
+
+            lineData.push({
+              headNumber: line?.headNumber,
+              baseNumber: line?.baseNumber,
+              headCoverage: line?.headCoverage,
+              baseCoverage: line?.baseCoverage,
+              hitCount: without(
+                line?.coverageInfo?.hitUploadIds,
+                ...ignoredUploadIds
+              ).length,
+            })
+          })
+        }
```

Comment:
could we move this into a useMemo? We seem to be doing a lot of work here

Reply:
Hrrm, this will probably require a pretty significant re-working of how the diff viewers work (I've also already merged on) due to the nested nesting of loops. Currently this is featured flagged and not turned on for any users, so what I'll do is come back around once all of them are merged and come up with a common implementation that resolves this issue fully.

Reply:
(may also throw in a TS refactor while I'm at it)

Reply:
sounds good to me. Yeah, I was typing out that comment and was thinking "this could be a lot of refactoring with how it ties in with the rest of the render function"